File: Test/m/math_reduction.m
function [vector_norm, log_and, log_or, bit_and, bit_or] = math_reductions(a, b, v)
    % max_global = max(max(a));
    % max_col = max(a);
    % max_row = max(a, [], 2);
    % max_elem_wise = max(a, b);
    vector_norm = norm(v);
    log_and = a & b;
    log_or = a | b;
    bit_and = bitand(a, b);
    bit_or = bitor(a, b);
end

File: Test/py/math_reduction.py
from numpy import *
import scipy.linalg as linalg

def math_reductions(a, b, v):
    # max_global = a.max()
    # max_col = a.max(0)
    # max_row = a.max(1)
    # max_elem_wise = maximum(a, b)
    vector_norm = linalg.norm(v)
    log_and = logical_and(a, b)
    log_or = logical_or(a, b)
    bit_and = a & b
    bit_or = a | b
    return vector_norm, log_and, log_or, bit_and, bit_or